### DISCUSSION TITLE
LC0012 - Include check on hardcoded object id in CONST expression on properties

### DISCUSSION THREAD

**Discussion by fvet** [StefanMaron/BusinessCentral.LinterCop#213] [New Rule]
Following the upcoming changes in BC21 wave 2, it would be nice if we could have a new rule or have rule LC0012 updated, to also warn on object id's usage (integer values) as part of a CONST expression in certain properties:

- SourceTableView 
- TableRelation
- SubPageLink
- _RunPageLink (not supported yet)_

https://yzhums.com/29459/

PS: Some warnings would be false positives and will require pragma directive, since the CONST expression in the properties are not exclusive used for object id's (can contain other integer which do not reference object id's)

![image](https://user-images.githubusercontent.com/25268332/190126803-f38c6e2f-d785-4df8-be3f-64ef368bbc95.png)

**Comment by StefanMaron** [comment]
Yep, had that on my list as well, thanks for providing the details! I think I will update the existing rule
